src/ring_theory/coprime.lean

```diff
@@ -166,10 +169,32 @@ by { rw [← finset.card_range n, ← finset.prod_const], exact is_coprime.prod_
 theorem is_coprime.pow (H : is_coprime x y) : is_coprime (x ^ m) (y ^ n) :=
 H.pow_left.pow_right
 
+theorem is_coprime.of_pow_left (hm : 0 < m) (h : is_coprime (x ^ m) y) : is_coprime x y :=
+begin
+  rw [← finset.card_range m, ← finset.prod_const] at h,
+  exact h.of_prod_left 0 (finset.mem_range.mpr hm),
+end
+
+theorem is_coprime.of_pow_right (hm : 0 < m) (h : is_coprime x (y ^ m)) : is_coprime x y :=
+(h.symm.of_pow_left hm).symm
+
+theorem is_coprime.of_pow (hm : 0 < m) (hn : 0 < n) (h : is_coprime (x ^ m) (y ^ n)) :
+  is_coprime x y :=
+(h.of_pow_left hm).of_pow_right hn
+
 theorem is_coprime.is_unit_of_dvd (H : is_coprime x y) (d : x ∣ y) : is_unit x :=
 let ⟨k, hk⟩ := d in is_coprime_self.1 $ is_coprime.of_mul_right_left $
 show is_coprime x (x * k), from hk ▸ H
 
+theorem is_coprime.is_unit_of_dvd' {a b x : R} (h : is_coprime a b) (ha : x ∣ a) (hb : x ∣ b) :
+  is_unit x :=
+begin
+  rw ←is_coprime_self,
+  obtain ⟨a', rfl⟩ := ha,
+  obtain ⟨b', rfl⟩ := hb,
+  exact h.of_mul_left_left.of_mul_right_left,
+end
+
```

Comment:
if you add the lemma `coprime b c -> a | b -> coprime a c` (and its symmetric counterpart) then you can refactor this proof into a one-liner using `is_coprime.is_unit_of_dvd`.